src/packages/files/files.ts

```diff
@@ -89,6 +90,59 @@ export default (app: Express) => {
     }
   });
 
+  /**
+   * @swagger
+   * /file/{id}:
+   *   get:
+   *     parameters:
+   *       - description: ID of the file
+   *         in: path
+   *         name: id
+   *         required: true
+   *         schema:
+   *           type: number
+   *     produces:
+   *       - application/json
+   *     responses:
+   *       200:
+   *         description: Fetches the file by id
+   *     summary: Get the file with the associated ID.
+   *     tags:
+   *       - Files
+   */
+  app.get('/file/:id', filesLimiter, auth, async (req, res) => {
+    const { params } = req;
+    const { id } = params;
+
+    try {
+      const { rows, rowCount } = await db.query(
+        `
+        select *
+        from merlin.uploaded_file
+        where id = $1
+        and deleted_date is null;
+      `,
+        [id],
+      );
+
+      if (rowCount && rowCount > 0) {
+        const [row] = rows;
+        const { name } = row;
+
+        const filePath = path.join(FILE_PATH, name);
+
+        res.sendFile(filePath, err => {
+          if (err) {
+            logger.info(`GET /file/{id}: No file was found with id: ${id}`);
+          }
+        });
```

Comment:
The endpoint should return a 404 response if the file isn't found, not a 200 with an error message.

Reply:
Okay, I see you're doing a try-catch structure, and that this is a callback function if something goes wrong. However, I would suggest that the top-level `catch` returns a 500 (internal error) code, and the callback function returns the 404 on error. That way, if something goes wrong further up the function (ie, there's a DB connection issue on L118), the endpoint will not return an inaccurate "Resource Not Found" error code.